qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+    and `c` is a character.
+    Allowed characters in `term_str` are 'N+-I`.
+    The returned tuple contains counts for `N`, `+`, and `-`, in that order. Identity operators
+    are ignored.
+
+    Args:
+       `term_str`: a string of characters in `+-NI`.
+       `expand_number_op`: if `True`, number operators are expanded to `(i, '+')`, `(i, '-')`
+         in the returned list of factors.
+
+    Returns:
+       tuple: A tuple of two elements. First, a tuple of three integers giving the number of
+       number-, raising-, and -lowering operators. Second a list of factors represented by
+       tuples of two elements: the first is an index and the second one of "-", "+", or "N".
+       If `expand_number_op` is `True`, then factors of `N` are expanded.
+
+    Raises:
+       ValueError: if any character in `term_str` is not one of "+-IN".
+    """
+    (n_number, n_raise, n_lower) = (0, 0, 0)
+    facs = []
+    for i, c in enumerate(term_str):
+        if c == "I":
+            continue
+        if c == "+":
+            n_raise += 1
+            facs.append((i, "+"))
+        elif c == "-":
+            n_lower += 1
+            facs.append((i, "-"))
+        elif c == "N":
+            n_number += 1
+            if expand_number_op:
+                facs.append((i, "+"))
+                facs.append((i, "-"))
+            else:
+                facs.append((i, "N"))
+        else:
+            raise ValueError("Unexpected operator ", c, " in term.")
+
+    return (n_number, n_raise, n_lower), facs
+
+
+def _interaction_type(n_number: int, n_raise: int, n_lower: int) -> TermType:
+    """
+    Return a `TermType` instance describing the type of interaction given the number of
+    number, raising, and lowering operators.
+
+    Args:
+       `n_number`: the number of number operators
+       `n_raise`: the number of raising operators
+       `n_lower`: the number of lowering operators
+
+    Returns:
+      TermType: The type of interaction.
+
+    Raises:
+      ValueError: if the numbers of operators don't describe a one- or two-body term from
+      an electronic Hamiltonian.
+    """
+    if n_raise == 0 and n_lower == 0:
+        if n_number == 1:
+            return TermType.NUMBER
+        elif n_number == 2:
+            return TermType.COULOMB_EXCHANGE
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 1 and n_lower == 1:
+        if n_number == 1:
+            return TermType.NUMBER_EXCITATION
+        elif n_number == 0:
+            return TermType.EXCITATION
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 2 and n_lower == 2:
+        return TermType.DOUBLE_EXCITATION
+    else:
+        raise ValueError("unexpected number of operators: ", n_number)
+
+
+def _get_adjacency_matrix(fer_op: FermionicOp) -> np.ndarray:
```

Comment:
After a little investigating, it looks like this may be a feature that will land soon, or at best, very recently. Or, there may be a few packages and modules that you can install as a workaround. I'm inclined to wait until this feature is better supported.